controllers/tlspolicy_status.go

```diff
@@ -67,5 +75,88 @@ func (r *TLSPolicyReconciler) calculateStatus(tlsPolicy *v1alpha1.TLSPolicy, spe
 	meta.SetStatusCondition(&newStatus.Conditions, *acceptedCond)
 
+	// Do not set enforced condition if Accepted condition is false
+	if meta.IsStatusConditionFalse(newStatus.Conditions, string(gatewayapiv1alpha2.PolicyReasonAccepted)) {
+		meta.RemoveStatusCondition(&newStatus.Conditions, string(kuadrant.PolicyConditionEnforced))
+		return newStatus
+	}
+
+	enforcedCond := r.enforcedCondition(ctx, tlsPolicy, targetNetworkObject)
+	meta.SetStatusCondition(&newStatus.Conditions, *enforcedCond)
+
 	return newStatus
 }
+
+func (r *TLSPolicyReconciler) enforcedCondition(ctx context.Context, tlsPolicy *v1alpha1.TLSPolicy, targetNetworkObject client.Object) *metav1.Condition {
+	if err := r.isIssuerReady(ctx, tlsPolicy); err != nil {
+		return kuadrant.EnforcedCondition(tlsPolicy, kuadrant.NewErrUnknown(tlsPolicy.Kind(), err), false)
+	}
+
+	if err := r.isCertificatesReady(ctx, tlsPolicy, targetNetworkObject); err != nil {
+		return kuadrant.EnforcedCondition(tlsPolicy, kuadrant.NewErrUnknown(tlsPolicy.Kind(), err), false)
+	}
+
+	return kuadrant.EnforcedCondition(tlsPolicy, nil, true)
+}
+
+func (r *TLSPolicyReconciler) isIssuerReady(ctx context.Context, tlsPolicy *v1alpha1.TLSPolicy) error {
+	var conditions []certmanv1.IssuerCondition
+
+	switch tlsPolicy.Spec.IssuerRef.Kind {
+	case "", certmanv1.IssuerKind:
+		issuer := &certmanv1.Issuer{}
+		if err := r.Client().Get(ctx, client.ObjectKey{Name: tlsPolicy.Spec.IssuerRef.Name, Namespace: tlsPolicy.Namespace}, issuer); err != nil {
+			return err
+		}
+		conditions = issuer.Status.Conditions
+	case certmanv1.ClusterIssuerKind:
+		issuer := &certmanv1.ClusterIssuer{}
+		if err := r.Client().Get(ctx, client.ObjectKey{Name: tlsPolicy.Spec.IssuerRef.Name}, issuer); err != nil {
+			return err
+		}
+		conditions = issuer.Status.Conditions
+	default:
+		return fmt.Errorf(`invalid value %q for issuerRef.kind. Must be empty, %q or %q`, tlsPolicy.Spec.IssuerRef.Kind, certmanv1.IssuerKind, certmanv1.ClusterIssuerKind)
+	}
+
+	transformedCond := utils.Map(conditions, func(c certmanv1.IssuerCondition) metav1.Condition {
+		return metav1.Condition{Reason: c.Reason, Status: metav1.ConditionStatus(c.Status), Type: string(c.Type), Message: c.Message}
+	})
+
+	if meta.IsStatusConditionFalse(transformedCond, string(certmanv1.IssuerConditionReady)) {
+		return errors.New("issuer not ready")
+	}
+
+	return nil
+}
+
+func (r *TLSPolicyReconciler) isCertificatesReady(ctx context.Context, tlsPolicy *v1alpha1.TLSPolicy, targetNetworkObject client.Object) error {
+	gwDiffObj, err := reconcilers.ComputeGatewayDiffs(ctx, r.Client(), tlsPolicy, targetNetworkObject)
+	if err != nil {
+		return err
+	}
+
+	if len(gwDiffObj.GatewaysWithValidPolicyRef) == 0 {
+		return errors.New("no valid gateways found")
+	}
+
+	for _, gw := range gwDiffObj.GatewaysWithValidPolicyRef {
+		expectedCertificates := r.expectedCertificatesForGateway(ctx, gw.Gateway, tlsPolicy)
+
+		for _, cert := range expectedCertificates {
```

Comment:
Not sure if this is what we want but there's no error if `len(expectedCerificates) == 0` since this is already allowed at:
- https://github.com/Kuadrant/kuadrant-operator/blob/bca513031565aafea05cb8ea0469cacc906e369a/controllers/tlspolicy_certmanager_certificates.go#L115-L119
- https://github.com/Kuadrant/kuadrant-operator/blob/bca513031565aafea05cb8ea0469cacc906e369a/controllers/tlspolicy_certmanager_certificates.go#L36-L39

so I've kept it the same. In this case a TLSPolicy is still marked as `enforced: true` even if there are no `Certificates` created

Reply:
I think that is ok. If there are no expected certificates then we have done everything we can to enforce